Storage/Processor/Processor.php

```diff
@@ -219,14 +219,19 @@ public function getResource(Config $config, string $filename)
             return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
         }
 
-        try {
-            return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
-        } catch (NotFoundException $e) {
-            $generatedResource = $this->generateResource($config);
-            if ($cacheableResult) {
-                $this->storageManager->cacheResource($generatedResource, $config->getAssetHash(), $config->getConfigHash(), $filename, $config->getMimeType());
+        $cacheableResult = $config->cacheableResult();
+        if ($cacheableResult && !$noCache) {
+            try {
+                return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
+            } catch (NotFoundException $e) {
+                //we have tried to get a cached version matching this request
```

Comment:
If this comment is usefull, I would transform it to a log message. A comment is lost in general (and not always maintained on refactoring)

Reply:
I refactor in order to clarify the cpode (and remove the comment)